controllers/kuadrant_controller.go

```diff
@@ -386,14 +392,26 @@ func (r *KuadrantReconciler) createOnlyInKuadrantNSCb(ctx context.Context, kObj
 			return err
 		}
 
-		k8sObjKind := k8sObj.DeepCopyObject().GetObjectKind()
+		var newObj client.Object
+		newObj = k8sObj
+
+		switch obj := k8sObj.(type) {
+		case *appsv1.Deployment: // If it's a Deployment obj, it adds the required env vars
+			obj.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+				{Name: envLimitadorNamespace, Value: kObj.Namespace},
+				{Name: envLimitadorName, Value: limitadorName},
+			}
+			newObj = obj
+		default:
+		}
+		err = r.Client().Create(ctx, newObj)
 
-		err = r.Client().Create(ctx, k8sObj)
-		logger.V(1).Info("create resource", "GKV", k8sObjKind.GroupVersionKind(), "name", k8sObj.GetName(), "error", err)
+		k8sObjKind := newObj.DeepCopyObject().GetObjectKind()
```

Comment:
deep copy before creating the object. For an unknown reason, the go client removes type info of the object being passed to the `Create` method.